File: 266-PalindromePermutation/266-PalindromePermutation.java
// Last updated: 25/03/2026, 21:42:44
class Solution {
    public boolean canPermutePalindrome(String s) {
        Set<Character> stringSet = new HashSet();

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (!stringSet.add(c)) {
                stringSet.remove(c);
            }
        }

        return stringSet.size() <= 1;
    }
}

File: 266-PalindromePermutation/266-PalindromePermutation.js
// Last updated: 25/03/2026, 21:42:39
/**
 * @param {string} s
 * @return {boolean}
 */
var canPermutePalindrome = function(s) {
    if (!s) return true;

    const stringSet = new Set();
    for (const letter of s) {
        if (stringSet.has(letter)) stringSet.delete(letter);
        else stringSet.add(letter);
    }

    return stringSet.size <= 1;
};